libsolidity/analysis/TypeChecker.cpp

```diff
@@ -639,6 +639,15 @@ void TypeChecker::visitManually(
 						"Can only use modifiers defined in the current contract or in base contracts."
 					);
 			}
+		if (
+			*_modifier.name().annotation().requiredLookup == VirtualLookup::Static &&
+			!modifierDecl->isImplemented()
+		)
+			m_errorReporter.typeError(
+				1835_error,
+				_modifier.location(),
+				"Cannot call unimplemented modifier when referencing it via its contract name."
```

Comment:
This sounds a bit as if you could call an unimplemented modifier if you referenced it differently. How about this?
```suggestion
				"Cannot call unimplemented modifier. The modifier has no implementation in the referenced contract."
```

Reply:
I wanted to make it clear that you can call the modifier when referencing it without prefixing the contract name.

Reply:
So maybe this?
```suggestion
				"Cannot call unimplemented modifier. The modifier has no implementation in the referenced contract. Refer to it by its unqualified name if you want to call the implementation from the most derived contract."
```

Reply:
At first I thought the text is way too long, but probably better to explain more than too little.

Reply:
Maybe we should have something like extra hints for errors. This way the main message could be short and to the point while the user could request a more verbose version with extra explanations.